packages/stream_video_noise_cancellation/lib/noise_cancellation_audio_processor.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:stream_video/stream_video.dart';
+import 'package:stream_video_noise_cancellation/stream_video_noise_cancellation.dart';
+
+class NoiseCancellationAudioProcessor extends AudioProcessor {
+  final _logger = taggedLogger(tag: 'SV:NoiseCancellationAudioProcessor');
+
+  NoiseCancellationAudioProcessor() {
+    try {
+      StreamVideoNoiseCancellation().registerProcessor();
+    } on UnimplementedError catch (e) {
+      _logger.w(
+        () => 'Noise cancellation is not supported on this platform: $e',
+      );
+    }
+  }
+
+  @override
+  Future<Result<bool>> isEnabled() async {
+    try {
+      await StreamVideoNoiseCancellation().isEnabled() ?? false;
+      return Result.success(true);
```

Comment:
You're always returning `true` here if you don't get an error.
```suggestion
      final result = await StreamVideoNoiseCancellation().isEnabled() ?? false;
      return Result.success(result);
```